Source/DafnyCore/AST/Node.cs

```diff
@@ -134,6 +77,78 @@ public abstract class Node : INode {
     }
     set => rangeToken = value;
   }
+}
+
+public abstract class RangeNode : Node {
+  public override RangeToken RangeToken { get; set; } // TODO remove set when TokenNode is gone.
+
+  protected RangeNode(Cloner cloner, RangeNode original) {
+    RangeToken = cloner.Tok(original.RangeToken);
+  }
+
+  protected RangeNode(RangeToken rangeToken) {
+    RangeToken = rangeToken;
+  }
+}
+
+
+public abstract class Node : INode {
```

Comment:
Can you please move this class above its children? I would like it better.

Reply:
Of course, no problem :)